api_blogs/lib/api_blogs/blog.ex

```diff
@@ -298,21 +314,15 @@ defmodule ApiBlogs.Blog do
     Post.changeset(post, attrs)
   end
 
-  @doc """
-  User login.
-  """
-  def do_login(%{"email" => "", "password" => _password}), do: {:error, :bad_request, "\"email\" is not allowed to be empty"}
-  def do_login(%{"email" => _email, "password" => ""}), do: {:error, :bad_request, "\"password\" is not allowed to be empty"}
-  def do_login(%{"email" => email, "password" => password}) do
-    User
-    |> Repo.get_by(email: email)
-    |> validate_login(password)
+  def search_posts_by_term(searchTerm) do
+    query =
+      from p in Post,
+      where: ilike(p.title, ^"%#{searchTerm}%")
+        or ilike(p.content, ^"%#{searchTerm}%")
+    result = Repo.all(query)
+    case result do
+      [] -> []
+      _ -> get_post_user(result)
+    end
```

Comment:
Aqui da pra usar os pipes pra encadear tudo que tal ?
```suggestion
    from p in Post,
     where: ilike(p.title, ^"%#{searchTerm}%")
       or ilike(p.content, ^"%#{searchTerm}%")
    |> Repo.all()
    |> case  do
      [] -> []
      _ -> get_post_user(result)
    end
```

Reply:
Obrigada pela ideia, ainda não acostumei a colocar pipe em todo lugar possível.

Reply:
@Caciquez essa mudança gerou um erro de compilação:

```
== Compilation error in file lib/api_blogs/blog.ex ==
** (CompileError) lib/api_blogs/blog.ex:318: the function "from" cannot handle clauses with the -> operator because it is not a macro. Please make sure you are invoking the proper name and that it is a macro
    lib/api_blogs/blog.ex:317: (module)
```